### PR TITLE
PF-2499: Handling for group constraint merging

### PR DESCRIPTION
**Pull request by mikenev**
In milestone 1, merging group constraints should only be possible if:
- The group list of the source and destination match exactly
- or the source doesn't have a group constraint.

Otherwise the merge should fail.

### CODE REVIEW

File: service/src/test/java/bio/terra/policy/service/policy/PolicyRegionConstraintTest.java [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package bio.terra.policy.service.policy;
 
+import static bio.terra.policy.service.policy.PolicyTestUtils.*;
```
Comment (ddietterich) [inline service/src/test/java/bio/terra/policy/service/policy/PolicyRegionConstraintTest.java:3]: No `*` imports. You can prevent that with IntelliJ settings.

File: service/src/test/java/bio/terra/policy/service/policy/PolicyGroupConstraintTest.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
+package bio.terra.policy.service.policy;
+
+import static bio.terra.policy.service.policy.PolicyTestUtils.*;
```
Comment (ddietterich) [inline service/src/test/java/bio/terra/policy/service/policy/PolicyGroupConstraintTest.java:3]: No `*` imports.

File: service/src/test/java/bio/terra/policy/controller/TpsExplainControllerTest.java [hunk 3]
```diff
@@ -17,21 +17,19 @@
 import bio.terra.policy.generated.model.ApiTpsPolicyInput;
 import bio.terra.policy.generated.model.ApiTpsPolicyPair;
 import bio.terra.policy.testutils.TestUnitBase;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.UUID;
+import java.util.*;
```
Comment (zloery) [inline service/src/test/java/bio/terra/policy/controller/TpsExplainControllerTest.java:20]: There are still some `*` imports after the latest commit

File: service/src/test/java/bio/terra/policy/controller/TpsExplainControllerTest.java [hunk 4]
```diff
@@ -49,8 +47,9 @@ public class TpsExplainControllerTest extends TestUnitBase {
           .name(GROUP_CONSTRAINT)
           .addAdditionalDataItem(MN_POLICY_PAIR);
 
+  // TODO: PF-2503 using MNGROUP for YU_POLICY_PAIR since dissimilar groups cannot merge.
   private static final ApiTpsPolicyPair YU_POLICY_PAIR =
-      new ApiTpsPolicyPair().key(GROUP).value(YUGROUP);
+      new ApiTpsPolicyPair().key(GROUP).value(MNGROUP);
```
Comment (zloery) [inline service/src/test/java/bio/terra/policy/controller/TpsExplainControllerTest.java:52]: Could this still be `YUGROUP` given they're the same underlying string?

**Review by zloery — APPROVED** [review]

**Pull request merged**